allennlp/models/semantic_parsing/nlvr/nlvr_decoder_step.py

```diff
@@ -401,31 +393,17 @@ def _compute_new_states(cls,
                                          attended_sentence[group_index],
                                          state.rnn_state[group_index].encoder_outputs,
                                          state.rnn_state[group_index].encoder_output_mask)
-                if new_checklist is None:
-                    new_state = NlvrDecoderState(batch_indices=[batch_index],
-                                                 action_history=[new_action_history],
-                                                 score=[new_score],
-                                                 rnn_state=[new_rnn_state],
-                                                 grammar_state=[new_grammar_state],
-                                                 action_embeddings=state.action_embeddings,
-                                                 action_indices=state.action_indices,
-                                                 possible_actions=state.possible_actions,
-                                                 worlds=state.worlds,
-                                                 label_strings=state.label_strings)
-                else:
-                    new_state = NlvrDecoderState(batch_indices=[batch_index],
-                                                 action_history=[new_action_history],
-                                                 score=[new_score],
-                                                 rnn_state=[new_rnn_state],
-                                                 grammar_state=[new_grammar_state],
-                                                 action_embeddings=state.action_embeddings,
-                                                 action_indices=state.action_indices,
-                                                 possible_actions=state.possible_actions,
-                                                 worlds=state.worlds,
-                                                 label_strings=state.label_strings,
-                                                 terminal_actions=[state.terminal_actions[group_index]],
-                                                 checklist_target=[state.checklist_target[group_index]],
-                                                 checklist_masks=[state.checklist_mask[group_index]],
-                                                 checklist=[new_checklist])
+                new_state_checklist_state = [new_checklist_state] if new_checklist_state is not None else None
```

Comment:
Here you also wouldn't have to special-case this, and make a variable with this crazy name =).

Reply:
Done.